libraries/SITL/SIM_Vicon.cpp

```diff
@@ -204,6 +204,53 @@ void Vicon::update_vicon_position_estimate(const Location &loc,
             NULL, 0);
         msg_buf[msg_buf_index].time_send_us = time_send_us;
     }
+
+    // TO DO add Confidence Level
+    // # Confidence level value from T265: 0-3, remapped to 0 - 100: 0% - Failed / 33.3% - Low / 66.6% - Medium / 100% - High
+    // current_confidence_level = data.tracker_confidence * 33.33333f
+
+    // determine time, position, and angular deltas
+    uint64_t time_delta = now_us + time_offset_us - _pos_delta_prev_time_us;
+
+    Quaternion curr_quat;                   // Rotation to Frame_aerobody_current from Frame_earth_NED
+    curr_quat.from_euler(roll, pitch, yaw); // Rotation to Frame_earth_NED from Frame_aerobody_current
+    curr_quat = curr_quat.inverse();
```

Comment:
a slight simplification here is to change this line to "curr_quad.invert()".

Reply:
Thanks! I didn't know about that one!